src/event/AsyncEvent.php

```diff
@@ -34,20 +33,11 @@
  * When all the promises of a priority level have been resolved, the next priority level is called.
  */
 interface AsyncEvent{
-	/**
-	 * Add a promise to the set of promises that will be awaited before the next priority level is called.
-	 *
-	 * @phpstan-param Promise<null> $promise
-	 */
-	public function addPromise(Promise $promise) : void;
-
 	/**
 	 * Be prudent, calling an event asynchronously can produce unexpected results.
 	 * During the execution of the event, the server, the player and the event context may have changed state.
 	 *
-	 * @phpstan-param ObjectSet<Promise<null>> $promiseSet
-	 *
 	 * @phpstan-return Promise<null>
 	 */
-	public static function callAsync(AsyncEvent&Event $event, ObjectSet $promiseSet) : Promise;
+	public function callAsync() : Promise;
```

Comment:
Does this mean the same event class can be called either synchronously or asynchronously? What happens if PlayerChatEvent is dispatched with `call()` in another plugin?

Reply:
Currently, both are possible.
I was wondering whether an exception should be thrown when someone tries to call an async event in sync.  
As a result, if an async event is called in sync, all the handlers are called without taking into account the return types, i.e. without waiting for the promises of the others, etc...

Reply:
It is no longer possible to call an asynchronous event synchronously.